teamsNotifyUser does not accept messages returned from MessageFactory.

### [Github issues](https://github.com/Microsoft/botbuilder-js/issues) should be used for bugs and feature requests. Use [Stack Overflow](https://stackoverflow.com/questions/tagged/botframework) for general "how-to" questions. 

## Versions
What package version of the SDK are you using: 1.15.0
What nodejs version are you using: 16.11.0
What browser version are you using: n/a, this is a typescript compiler error
What os are you using: Win 11, 22H2, 22567.200

## Describe the bug
When using teamsNotifyUser in a typescript application [the example provided by the docs](https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/conversation-messages?tabs=typescript#notifications-to-your-message) on how to notify someone does not work.
Instead, the following error is returned:
```
Argument of type 'Partial<Activity>' is not assignable to parameter of type 'Activity'.
  Property 'type' is optional in type 'Partial<Activity>' but required in type 'Activity'.ts(2345)
```

This is because `MessageFactory.text('Message');` returns a `Partial<Activity>` while `teamsNotifyUser()` requires a straight `Activity`.

## To Reproduce

1. In Teams Toolkit, create a new app. Selecting only Bot as a capability and Typescript as the language.
2. In `bot/teamsBot.ts` replace
```ts
case "welcome": {
    const card = AdaptiveCards.declareWithoutData(rawWelcomeCard).render();
    await context.sendActivity({ attachments: [CardFactory.adaptiveCard(card)] });
    break;
}
```
with
```ts
case "welcome": {
    let message = MessageFactory.text("You'll get a notification, if you've turned them on.");
    teamsNotifyUser(message);
    await context.sendActivity(message);
}
```
3. From `./bot` run `npm run build`
4. See the following build error:
```
teamsBot.ts:39:27 - error TS2345: Argument of type 'Partial<Activity>' is not assignable to parameter of type 'Activity'.
  Property 'type' is optional in type 'Partial<Activity>' but required in type 'Activity'.

39           teamsNotifyUser(message);
``` 


## Expected behavior
To be able to call `teamsNotifyUser()` with the value returned from the MessageFactory.

## Screenshots
<img width="767" alt="image" src="https://user-images.githubusercontent.com/6508687/157708186-39e32ea2-8979-404d-a4bb-62563de289e6.png">
<img width="751" alt="image" src="https://user-images.githubusercontent.com/6508687/157712435-feb705bc-7ff4-4b66-9409-477855a62baa.png">

## Additional context
I debated creating a bug against the SDK or the docs but felt it made more sense to create it here.

--- a/libraries/botbuilder/etc/botbuilder.api.md
+++ b/libraries/botbuilder/etc/botbuilder.api.md
@@ -444,7 +444,7 @@ export class TeamsInfo {
 }
 
 // @public
-export function teamsNotifyUser(activity: Activity, alertInMeeting?: boolean, externalResourceUrl?: string): void;
+export function teamsNotifyUser(activity: Partial<Activity>, alertInMeeting?: boolean, externalResourceUrl?: string): void;
 
 // @public
 export class TeamsSSOTokenExchangeMiddleware implements Middleware {

--- a/libraries/botbuilder/src/teamsActivityHelpers.ts
+++ b/libraries/botbuilder/src/teamsActivityHelpers.ts
@@ -12,7 +12,7 @@ function isTeamsChannelData(channelData: unknown): channelData is TeamsChannelDa
     return typeof channelData === 'object';
 }
 
-function validateActivity(activity: Activity): void {
+function validateActivity(activity: Partial<Activity>): void {
     if (!activity) {
         throw new Error('Missing activity parameter');
     }
@@ -104,7 +104,11 @@ export function teamsGetChannelId(activity: Activity): string | null {
  * @param alertInMeeting Sent to a meeting chat, this will cause the Teams client to render it in a notification popup as well as in the chat thread.
  * @param externalResourceUrl Url to external resource. Must be included in manifest's valid domains.
  */
-export function teamsNotifyUser(activity: Activity, alertInMeeting?: boolean, externalResourceUrl?: string): void {
+export function teamsNotifyUser(
+    activity: Partial<Activity>,
+    alertInMeeting?: boolean,
+    externalResourceUrl?: string
+): void {
     validateActivity(activity);
 
     if (!isTeamsChannelData(activity.channelData)) {

